File: P120812.py
def solution(array):
    answer = 0
    arr = [0] * 1001
    for num in array:
        arr[num] += 1
    Max = max(arr)
    cnt = 0
    for i in range(1001):
        if arr[i] == Max:
            answer = i
            cnt += 1
    
    if cnt >= 2:
        answer = -1
        
    return answer

File: P120812.java
public class P120812 {
    class Solution {
        public int solution(int[] array) {
            int answer = 0;
            int max = 0;
            int[] count = new int[1001];
            for (int i=0; i<array.length; i++) count[array[i]]++;
            for (int i=0; i<1001; i++) {
                if (count[i] > max) {
                    max = count[i];
                    answer = i;
                }
            }
            
            int maxCount = 0;
            for (int i=0; i<1001; i++) {
                if (count[i] == max) maxCount += 1;
            }
            
            if (maxCount > 1) answer = -1;
            return answer;
        }
    }
}